### DISCUSSION TITLE
Flex Removal Impact

### DISCUSSION THREAD

**Discussion by abubakar2029** [tailwindlabs/tailwindcss#12487] [General]
import { Fragment, useState } from "react";
import { Dialog, Transition } from "@headlessui/react";
import { XMarkIcon } from "@heroicons/react/24/outline";

function Example() {
  const [open, setOpen] = useState(true);

  return (
    <>
      <button className="text-[32px] font-black" onClick={() => setOpen(true)}>
        Aho
      </button>
      <Transition.Root show={open} as={Fragment}>
        <Dialog as="div" className="relative z-10" onClose={setOpen}>
          <Transition.Child
            as={Fragment}
            enter="ease-in-out duration-500"
            enterFrom="opacity-0"
            enterTo="opacity-100"
            leave="ease-in-out duration-500"
            leaveFrom="opacity-100"
            leaveTo="opacity-0"
          >
            <div className="fixed inset-0 bg-gray-500 bg-gradient-to-bl from-orange-500 from-30% via-purple-400 via-40% to-orange-300" />
          </Transition.Child>
          <div className="fixed inset-0 overflow-hidden">
            {/* <div className="absolute inset-0 overflow-hidden"> */}
              <div className="pointer-events-none flex fixed inset-y-0 right-0  bg-red-500  max-w-full pl-10">
                <Transition.Child
                  as={Fragment}
                  enter="transform transition ease-in-out duration-500 sm:duration-700"
                  enterFrom="translate-x-full"
                  enterTo="translate-x-0"
                  leave="transform transition ease-in-out duration-500 sm:duration-700"
                  leaveFrom="translate-x-0"
                  leaveTo="translate-x-full"
                >
                  <Dialog.Panel className="pointer-events-auto relative bg-blue-500 p-4 w-screen max-w-md">
                    <Transition.Child
                      as={Fragment}
                      enter="ease-in-out duration-500"
                      enterFrom="opacity-0"
                      enterTo="opacity-100"
                      leave="ease-in-out duration-500"
                      leaveFrom="opacity-100"
                      leaveTo="opacity-0"
                    >
                      <div className="absolute left-0 top-0 -ml-8 flex pr-2 pt-4 sm:-ml-10 sm:pr-4">
                        <button
                          type="button"
                          className="relative rounded-md text-gray-300 hover:text-white focus:outline-none focus:ring-2 focus:ring-white"
                          onClick={() => setOpen(false)}
                        >
                          <span className="absolute -inset-2.5" />
                          <span className="sr-only">Close panel</span>
                          <XMarkIcon className="h-6 w-6" aria-hidden="true" />
                        </button>
                      </div>
                    </Transition.Child>
                    <div className="flex h-full flex-col overflow-y-scroll bg-white py-6 shadow-xl">
                      <div className="px-4 sm:px-6">
                        <Dialog.Title className="text-base font-semibold leading-6 text-gray-900">
                          Panel title
                        </Dialog.Title>
                      </div>
                      <div className="relative mt-6 flex-1 px-4 sm:px-6">
                        {/* Your content */}
                      </div>
                    </div>
                  </Dialog.Panel>
                </Transition.Child>
              </div>
            {/* </div> */}
          </div>
        </Dialog>
      </Transition.Root>
    </>
  );
}
export default Example;

why on removing flex from line:28(div with red background) changes height of it's child i.e. Dialog.Panel(bg-blue-500).
How flex is affecting n it's height.
i never read about such behavior before

**Comment by wongjn** [comment]
I don't see any `<div>` elements in your code. Please consider formatting your code properly within a codeblock or replicating the unexpected behavior in [Tailwind Play](https://play.tailwindcss.com).
